### ISSUE TITLE
Loading perspectives upon opening emacs

### ISSUE THREAD

**Issue by ikbenlike** [nex3/perspective-el#200]
As is demonstrated in the README, I have a hook to save perspectives when emacs is closed, like so:
```elisp
(add-hook 'kill-emacs-hook #'persp-state-save)
```
However, I would also like to automatically restore perspectives when I open emacs. I have tried adding `persp-state-load` to `window-setup-hook`, `after-init-hook`, `before-init-hook` and `emacs-startup-hook`. I have also tried manually calling it in my `init.el`. However, none of these seem to actuallly call `persp-state-load` - at least if they do, none of my perspectives actually get loaded, yet it exits without erroring. 

I am unsure if I am going about this incorrectly or if this is even something supported by `perspective.el`, though I do feel the README could benefit from clarification in this area.

**Comment by gcv** [comment]
`(persp-state-load "/path/to/persp-state-file.el")` should work from a startup file. There was a PR for giving `persp-state-load` default argument values which would make it work from a hook (https://github.com/nex3/perspective-el/pull/120), but it was never taken over the finish line.

**Comment by ikbenlike** [comment]
I tried `(persp-state-load persp-state-default-file)` in my `init.el`, but it doesn't seem to load the saved perspectives. It also doesn't seem to error at all, so it's hard for me to tell what's wrong. Loading the perspectives with keybinds still works in either case.

Also, as far as I can tell from [the code here in the repo](https://github.com/nex3/perspective-el/blob/master/perspective.el#L2029), `persp-state-load` does take a default value similar to `persp-state-save`, but I'm not very good at elisp so that might not be correct

**Comment by gcv** [comment]
`persp-state-load` only takes a default when called interactively (i.e. with a keybinding or from <kbd>M-x</kbd>). The PR I linked above was an attempt to have it also take a default when called non-interactively (i.e. from Lisp code).

I wonder if `(persp-state-load persp-state-default-file)` failed for you in `init.el` because you called it before loading and initializing `perspective.el`? Please try the code from `eval-expression` (<kbd>M-:</kbd> in vanilla Emacs) after startup. That works for me.

**Comment by ikbenlike** [comment]
running it through `eval-expression` does indeed work - however, from my understanding, `perspective.el` would already be loaded by the time it tries to run that line in my `init.el`, since I can succesfully add the save hook on the previous line, like so:
```
(use-package perspective
  :bind
  ("C-x C-b" . persp-list-buffers)
  :custom
  (persp-mode-prefix-key (kbd "C-z"))
  :init
  (persp-mode))

(add-hook 'kill-emacs-hook #'persp-state-save)
(persp-state-load persp-state-default-file)
```

**Comment by ikbenlike** [comment]
After some more testing I found out that I can auto-load the stored perspectives on startup in this way:
```
(add-hook 'after-init-hook
		  (lambda ()
			(persp-state-load persp-state-default-file)))
```
It appears simply putting the line `(persp-state-load persp-state-default-file)` at the end of the `init.el` file also works; however, I didn't notice because the default splash screen kept popping up. After I disabled the splash both of these options seem to behave the same. Two differences with `desktop.el` seem to be that `perspective.el` doesn't circumvent the splash and doesn't seem to save the emacs window size. The former is easily fixed, and the latter is not really a problem to me personally.
